src/plugins/intel_cpu/src/node.cpp

```diff
@@ -197,25 +197,22 @@ Node::Node(const std::string& type, const std::string& name, const GraphContext:
     // TODO [NM]: What about filling inDims and outDims?
 }
 
-void Node::addEdge(const EdgeWeakPtr& edge) {
-    auto edgePtr = edge.lock();
-    if (!edgePtr)
-        return;
-    auto parentPtr = edgePtr->getParent();
-    auto childPtr = edgePtr->getChild();
-    if (!parentPtr || !childPtr)
-        return;
+void Node::addEdge(const EdgePtr& edge) {
+    auto parent = edge->getParent();
+    auto child = edge->getChild();
+    assert(parent && child);
 
-    parentPtr->addChildEdge(edge);
-    childPtr->addParentEdge(edge);
+    parent->addChildEdge(edge);
+    child->addParentEdge(edge);
 }
 
 void Node::remove() {
     auto drop = [](std::vector<EdgeWeakPtr> edges){
         for (auto& edge : edges) {
             auto edgePtr = edge.lock();
             if (!edgePtr) continue;
-            edgePtr->drop();
```

Comment:
Do we need to keep `drop` method now?

Reply:
Will be removed

Reply:
Forgot to clarify, the motivation was to avoid logic where edge removes itself from the nodes, basically using **this** ptr. It really should be a graph responsibility.

Reply:
Done